capstone-backend/src/main/java/com/google/sps/data/Group.java

```diff
@@ -0,0 +1,117 @@
+package com.google.sps.data;
+
+import java.util.ArrayList;
+import java.util.List;
+
+ 
+public abstract class Group extends BaseEntity {
+  
+  private String name;
+  private String description;
+  private String announcement;
+  private List<String> posts;
+  private String ownerID;
+  private List<String> memberIDs;
+  private List<String> inviteIDs;
+
+  public Group(String name, String description, String announcement, List<String> posts, 
+      String ownerID, List<String> memberIDs, List<String> inviteIDs) {
+        this.name = name;
+        this.description = description;
```

Comment:
I don't think we need setters for this object. We should need to change the object once its constructed; the doPut will handle that.

Reply:
I think under the hood firebase uses public getters and setters for much of its functionality, so the doPut needs these setters in order to work.

Reply:
Yeah this is what I was thinking. If it's necessary for firestore I'll keep it here